pymc/step_methods/hmc/hmc.py

```diff
@@ -39,27 +39,25 @@ class HamiltonianMC(BaseHMC):
 
     name = "hmc"
     default_blocked = True
-    stats_dtypes = [
-        {
-            "step_size": np.float64,
-            "n_steps": np.int64,
-            "tune": bool,
-            "step_size_bar": np.float64,
-            "accept": np.float64,
-            "diverging": bool,
-            "energy_error": np.float64,
-            "energy": np.float64,
-            "path_length": np.float64,
-            "accepted": bool,
-            "model_logp": np.float64,
-            "process_time_diff": np.float64,
-            "perf_counter_diff": np.float64,
-            "perf_counter_start": np.float64,
-            "largest_eigval": np.float64,
-            "smallest_eigval": np.float64,
-            "warning": SamplerWarning,
-        }
-    ]
+    stats_dtypes_shapes = {
+        "step_size": (np.float64, []),
+        "n_steps": (np.int64, []),
+        "tune": (bool, []),
+        "step_size_bar": (np.float64, []),
+        "accept": (np.float64, []),
+        "diverging": (bool, []),
+        "energy_error": (np.float64, None),
+        "energy": (np.float64, None),
+        "path_length": (np.float64, None),
+        "accepted": (bool, []),
+        "model_logp": (np.float64, None),
+        "process_time_diff": (np.float64, None),
+        "perf_counter_diff": (np.float64, None),
+        "perf_counter_start": (np.float64, None),
+        "largest_eigval": (np.float64, None),
+        "smallest_eigval": (np.float64, None),
```

Comment:
These should be checked, and the `None`-shaped ones should be updated. Are they all scalar?

If there are stats for which the shape must be determined from the `vars`, we can update their entries from the `__init__`.